comp/snmpscan/def/component.go

```diff
@@ -0,0 +1,30 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2024-present Datadog, Inc.
+
+// Package snmpscan is a light component that can be used to perform a scan or a walk of a particular device
+package snmpscan
+
+import (
+	"github.com/DataDog/datadog-agent/comp/aggregator/demultiplexer"
+	log "github.com/DataDog/datadog-agent/comp/core/log/def"
+	"github.com/gosnmp/gosnmp"
+	"go.uber.org/fx"
+)
+
+// team: network-device-monitoring
+
+// Requires defines the dependencies for the snmpscan component
+type Requires struct {
+	fx.In
+	Logger        log.Component
+	Demultiplexer demultiplexer.Component
+}
```

Comment:
We want to avoid referencing `"go.uber.org/fx"` or internal components requires.

You can refer the documentation for creating components here: https://datadoghq.dev/datadog-agent/components/creating-components/

TL;DR
- Move the `Requires` struct into the `impl` folder.
- replace `fx.In` with `compdef.In` which is a one to one. 

If you want to check how we do it on other components, you can check https://github.com/DataDog/datadog-agent/blob/main/comp/core/healthprobe/impl/healthprobe.go#L27-L28

Reply:
I've struggled a bit with test, could you maybe help me here?

If I do as you say, my test `TestSnmpScanComp` starts failing, fx is unable to create my component.
It fails with the error `snmpscanimpl.Requires (did you mean to Provide it?)` which is well, what I'm trying to build.

How would you go about fixing that? I've turned a bit in circles here

Reply:
Also: I realized there was some component liniting tool which was failing when I tried keeping the struct in `impl` and importing `fx.In`. That's great, it let me now I should import fx in the `impl` folder.

That said, it didn't complain about importing `fx` in the def folder. Is that an oversight with the linter?